### PR TITLE
add clique_percolation.jl

### PR DESCRIPTION
**Pull request by suzhu1988**
clique percolation is a method of uncovering the overlapping community structure of complex networks in nature and society

### CODE REVIEW

**Comment by jpfairbanks** [comment]
@suzhu1988 this looks great, can you add some test cases? You can follow the example in https://github.com/JuliaGraphs/LightGraphs.jl/blob/master/test/community/cliques.jl. We strive for 100% coverage of all code being covered by tests.

Also, should this code cite the original clique percolation [paper]( https://www.nature.com/articles/nature03607), or a later document?

**Comment by suzhu1988** [comment]
Thanks for your advises, I have add a test file and the original clique percolation paper

**Comment by jpfairbanks** [comment]
Sorry I forgot to add: "you need to add your test file to list of tests: https://github.com/JuliaGraphs/LightGraphs.jl/blob/f93fbf823ed234a98575c6e1169d5cefed08bc19/test/runtests.jl#L46"

After that I think it is ready.

File: src/community/clique_percolation.jl [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            end
+        end
+        if common_nodes >= k-1
+            add_edge!(h, i, j)
+        end
+    end
+    # reset status
+    for u in kcliques[i]
+      x[u] = false
+    end
```
Comment (sbromberger) [inline src/community/clique_percolation.jl]: `x[kcliques[i]] = false`

File: src/community/clique_percolation.jl [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  end
+  components = connected_components(h)
+  communities = [IntSet() for i=1:length(components)]
+  for (i,component) in enumerate(components), u in component, v in kcliques[u]
+    push!(communities[i], v)
+   end
```
Comment (sbromberger) [inline src/community/clique_percolation.jl]: I think we can get rid of a loop variable here, but not 100% sure.

**Comment by jpfairbanks** [comment]
Tests are failing with this log message.
```
INFO: Testing LightGraphs
Clique percolation: Error During Test
  Test threw an exception of type UndefVarError
  Expression: @inferred test_cliques(g, Array[[1, 2, 3], [1, 4, 5]])
  UndefVarError: clique_percolation not defined
  Stacktrace:
   [1] (::#test_cliques#82)(::LightGraphs.SimpleGraphs.SimpleGraph{Int64}, ::Array{Array,1}) at /home/travis/.julia/v0.6/LightGraphs/test/community/clique_percolation.jl:8
   [2] macro expansion at /home/travis/.julia/v0.6/LightGraphs/test/community/clique_percolation.jl:18 [inlined]
   [3] macro expansion at ./test.jl:860 [inlined]
   [4] anonymous at ./<missing>:?```

Which doesn't make sense because the function is added to the export statements.

**Comment by sbromberger** [comment]
> Tests are failing with this log message.

I fixed it. Because this is a trait function, it requires a "generic" function in order to be exported. `function clique_percolation end` is the solution.

**Comment by sbromberger** [comment]
Gah. I think I messed this up. Sorry. Could someone please reopen the PR?

**Pull request closed without merging**